src/components/tasks/TaskInfoOverlay.js

```diff
@@ -10,36 +10,39 @@ import ButtonIcon from '../icons/ButtonIcon';
 import '../base/Overlay.css';
 import Separator from '../base/Separator';
 
+import {useTasks} from "../../hooks/useDataContext";
 
-const TaskInfoOverlay = ({ task, onExit }) => {
+const TaskInfoOverlay = ({ task, onExit, onOpenEdit }) => {
+    const { taskList } = useTasks();
+    const editedTask = taskList.find((t) => t.id === task.id);
```

Comment:
Nice if it works! That doesn't use useEffect, so it's great.

As someone once said: *the lesser you use useEffect, the better it is*.